scripts/start.js

```diff
@@ -62,48 +70,67 @@ if (process.env.HOST) {
 
 // We attempt to use the default port but if it is busy, we offer the user to
 // run on a different port. `choosePort()` Promise resolves to the next free port.
-choosePort(HOST, DEFAULT_PORT)
-  .then((port) => {
-    if (port == null) {
-      // We have not found a port.
-      return;
-    }
-    const protocol = process.env.HTTPS === "true" ? "https" : "http";
-    const appName = "sourcecred";
-    const urls = prepareUrls(protocol, HOST, port);
-    // Create a webpack compiler that is configured with custom messages.
-    const compiler = createCompiler(webpack, config, appName, urls, useYarn);
-    // Load proxy config
-    const proxySetting = undefined;
-    const proxyConfig = prepareProxy(proxySetting, paths.appPublic);
-    // Serve webpack assets generated by the compiler over a web sever.
-    const serverConfig = createDevServerConfig(
-      proxyConfig,
-      urls.lanUrlForConfig
-    );
-    const devServer = new WebpackDevServer(compiler, serverConfig);
-    // Launch WebpackDevServer.
-    devServer.listen(port, HOST, (err) => {
-      if (err) {
-        return console.log(err);
-      }
-      if (isInteractive) {
-        clearConsole();
-      }
-      console.log(chalk.cyan("Starting the development server...\n"));
-      openBrowser(urls.localUrlForBrowser);
-    });
+async function main() {
+  const webpackPort = await choosePort(HOST, DEFAULT_WEBPACK_PORT);
+  const apiPort = await choosePort(HOST, DEFAULT_API_PORT);
 
-    ["SIGINT", "SIGTERM"].forEach(function(sig) {
-      process.on(sig, function() {
-        devServer.close();
-        process.exit();
-      });
+  if (webpackPort == null) {
+    console.error("Could not find a port for the Webpack server.");
+  }
+  if (apiPort == null) {
+    console.error("Could not find a port for the API server.");
+  }
+
+  const sourcecredDirectory =
+    process.env.SOURCECRED_DIRECTORY || path.join(os.tmpdir(), "sourcecred");
+  const apiServer = await new Promise(async (resolve, _unused_reject) => {
+    let server = apiApp(sourcecredDirectory).listen(apiPort, () => {
+      resolve(server);
     });
-  })
-  .catch((err) => {
-    if (err && err.message) {
-      console.log(err.message);
+  });
+  console.log(
+    chalk.green(`Server listening on port ${apiServer.address().port}.`)
+  );
+  console.log();
```

Comment:
?

Reply:
It prints an empty line, to give some space.